### PR TITLE
Loading indicator for ruleEditor

### PR DESCRIPTION
**Pull request by isaac73**
# Type of change? 

- [ ] Bug fix
- [x] New feature
- [ ] Enhancement
- [ ] Breaking change (breaks backward compatibility)

# Description, Context, Motivation 

Loading indicator for ruleEditor

**Checklist:**

- [ ] All tests passed
- [ ] The code follows the code style and conventions of this project
- [ ] The change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js [hunk 1]
```diff
@@ -110,24 +111,27 @@ export class RuleEditor extends LinkedComponent {
...
       if (this.props.rule) { // If rule object exist then update the existing rule
         this.subscription = TelemetryService.updateRule(this.props.rule.id, toNewRuleRequestModel(requestProps))
           .subscribe(
             (updatedRule) => {
               updateRule(updatedRule);
+              this.setState({ isPending: false });
               onClose();
             },
-            error => this.setState({ error })
+            error => this.setState({ error, isPending: false, })
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js]: nit: extra , at the end

File: src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js [hunk 2]
```diff
@@ -110,24 +111,27 @@ export class RuleEditor extends LinkedComponent {
...
       } else { // If rule object doesn't exist then create a new rule
         this.subscription = TelemetryService.createRule(toNewRuleRequestModel(requestProps))
           .subscribe(
             (createdRule) => {
               insertRule(createdRule);
+              this.setState({ isPending: false });
               onClose();
             },
-            error => this.setState({ error })
+            error => this.setState({ error, isPending: false, })
```
Comment (ppathan) [inline src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js]: nit: extra , at end

File: src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js [hunk 3]
```diff
@@ -352,17 +356,19 @@ export class RuleEditor extends LinkedComponent {
             </FormGroup>
           </Section.Content>
         </Section.Container>
-        <SummarySection>
+        <SummarySection indicator={isPending} indicatorSize="large" indicatorPattern="bar">
```
Comment (spryor) [inline src/components/pages/rules/flyouts/ruleEditor/ruleEditor.js]: You don't need to do this. You just need to modify the the css a bit. 

In the SummarySection component update the container class to have `flex-flow: row wrap;` Then the following will work.

```
<SummarySection>
  <Indicator size="medium" pattern="bar" />
  <SummaryCount>{devicesAffected}</SummaryCount>
  <SectionDesc>{t('rules.flyouts.ruleEditor.devicesAffected')}</SectionDesc>
</SummarySection>
```

File: src/components/shared/forms/styles/summarySection.scss [hunk 4]
```diff
@@ -21,6 +21,10 @@
     }
   }
 
+  .indicator-padding {
```
Comment (spryor) [inline src/components/shared/forms/styles/summarySection.scss]: also oneline

File: src/components/pages/devices/flyouts/deviceNew/deviceNew.scss [hunk 5]
```diff
@@ -7,6 +7,10 @@
   @include rem-font-size(14px);
   @include rem-fallback(padding, 12px, 0px);
 
+  .devices-new-content {
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceNew/deviceNew.scss]: oneline

File: src/components/pages/devices/flyouts/deviceNew/deviceNew.js [hunk 6]
```diff
@@ -317,74 +317,76 @@ export class DeviceNew extends LinkedComponent {
...
+                    <FormLabel>{t('devices.flyouts.new.deviceId.label')}</FormLabel>
+                    <Radio link={this.isGenerateIdLink} value={DeviceIdTypeOptions.manual.value}>
+                      <FormControl className="device-id" link={this.deviceIdLink} disabled={isGenerateId} type="text" placeholder={t(DeviceIdTypeOptions.manual.hintName)} />
+                    </Radio>
+                    <Radio link={this.isGenerateIdLink} value={DeviceIdTypeOptions.generate.value}>
+                      {t(DeviceIdTypeOptions.generate.labelName)}
+                    </Radio>
+                  </FormGroup>,
+                  <FormGroup key="authType">
+                    <FormLabel >{t(AuthTypeOptions.labelName)}</FormLabel>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceNew/deviceNew.js]: nit: extra space in `<FormLabel >`

File: src/components/pages/devices/flyouts/deviceNew/deviceNew.js [hunk 7]
```diff
@@ -317,74 +317,76 @@ export class DeviceNew extends LinkedComponent {
...
+                    <Radio link={this.authenticationTypeLink} value={AuthTypeOptions.symmetric.value}>
+                      {t(AuthTypeOptions.symmetric.labelName)}
+                    </Radio>
+                    <Radio link={this.authenticationTypeLink} value={AuthTypeOptions.x509.value}>
+                      {t(AuthTypeOptions.x509.labelName)}
+                    </Radio>
+                  </FormGroup>,
+                  <FormGroup key="authKeyType">
+                    <FormLabel >{t(AuthKeyTypeOptions.labelName)}</FormLabel>
```
Comment (spryor) [inline src/components/pages/devices/flyouts/deviceNew/deviceNew.js]: another space

**Pull request merged**